src/scheduler/workers/processListsBeforeAndDuringStart/helpers.ts

```diff
@@ -14,36 +12,66 @@ export function formatDate(date: Date = todayDateString) {
+  if (!event?.jsonData || !reminderType || !event) {
     return false;
   }
-  const subsequentEmailsForReminderType = {
-    sendOneMonthPostEmail: [
-      "sendOneMonthPostEmail",
-      "sendOneWeekPostEmail",
-      "sendOneDayPostEmail",
-      "sendStartPostEmail",
-    ],
-    sendOneWeekPostEmail: ["sendOneWeekPostEmail", "sendOneDayPostEmail", "sendStartedPostEmail"],
-    sendOneDayPostEmail: ["sendOneDayPostEmail", "sendStartedPostEmail"],
-    sendStartedPostEmail: ["sendStartedPostEmail"],
-    sendStartedProviderEmail: ["sendStartedProviderEmail"],
+
+  const subsequentEmailsForReminderType: Record<RemindersBeforeStartDate, ListAnnualReviewPostReminderType[]> = {
+    oneMonthBeforeStart: ["oneMonthBeforeStart", "oneWeekBeforeStart", "oneDayBeforeStart", "started"],
+    oneWeekBeforeStart: ["oneWeekBeforeStart", "oneDayBeforeStart", "started"],
+    oneDayBeforeStart: ["oneDayBeforeStart", "started"],
+    started: ["started"],
   };
+
   const subsequentEmails: string[] = subsequentEmailsForReminderType[reminderType];
-  let reminderHasBeenSent = false;
-  if ("createdAt" in event) {
-    reminderHasBeenSent = subsequentEmails.includes((event.jsonData as ListEventJsonData)?.reminderType as string);
-  } else {
-    reminderHasBeenSent = subsequentEmails.includes((event.jsonData as Record<string, string>)?.notes[0]);
-  }
+  const reminderHasBeenSent = subsequentEmails.includes(
+    event.annualReviewEmailType ?? (event.jsonData as Record<string, string>)?.reminderType
+  );
 
   if (reminderHasBeenSent) {
     logger.info(`Email has been sent before for ${reminderType} reminder type`);
     return true;
   }
   return false;
 }
+
+/**
+ * Determines whether `emailType` should be sent for `annualReviewReference`.
+ * All annual review audits or events are recorded with the {@link annualReviewReference} or `reference` from `{@link List.jsonData.currentAnnualReview.reference}`.
+ */
+export async function shouldSend(
+  emailType: AnnualReviewProviderEmailType,
+  listItemId: number,
+  annualReviewReference?: string
+): Promise<boolean> {
+  if (!annualReviewReference) {
+    logger.warn(
+      `shouldSend: Annual review reference was not supplied, could not look up whether ${emailType} should be sent to ${listItemId}.`
+    );
+    return false;
+  }
+  /**
+   * This query uses postgres enums. Enums in postgres are ordered so `>` operator can be used.
+   * Looks for events with `listItemId` and where an annualReviewEmailType > than `emailType` has occurred.
+   */
+  const result: Event[] | undefined = await prisma.$queryRaw`select * from "Event"
+         where "listItemId" = ${listItemId}
+           and "annualReviewEmailType" > ${emailType}::"AnnualReviewProviderEmailType"
+           and "jsonData"->>'reference' = '${annualReviewReference}'
+           and "type" = 'REMINDER'
+         order by "time" desc limit 1`;
```

Comment:
Is there a way to do the `>` on enums with a prisma query?

Reply:
I couldn't find it in the docs. I had a quick look in generated types but I couldn't see what I wanted. I'll take another look

Reply:
```.ts
 export type EnumAnnualReviewProviderEmailTypeNullableFilter<$PrismaModel = never> = {
    equals?: AnnualReviewProviderEmailType | EnumAnnualReviewProviderEmailTypeFieldRefInput<$PrismaModel> | null
    in?: Enumerable<AnnualReviewProviderEmailType> | ListEnumAnnualReviewProviderEmailTypeFieldRefInput<$PrismaModel> | null
    notIn?: Enumerable<AnnualReviewProviderEmailType> | ListEnumAnnualReviewProviderEmailTypeFieldRefInput<$PrismaModel> | null
    not?: NestedEnumAnnualReviewProviderEmailTypeNullableFilter<$PrismaModel> | AnnualReviewProviderEmailType | null
  }
```

this is all you can query by it seems. int and time fields have "gte" / "lte"